bouncer/blacklist/views.py

```diff
@@ -27,3 +31,29 @@ def get(self, request):
             response = {"kind": "ip", "value": ip, "result": result}
             result_list.append(response)
         return Response(result_list)
+
+
+class PostRequestViewSet(viewsets.ViewSet):
+    @action(methods=["post"], detail=True)
+    def add_entry(self, request, pk=None):
+        data = json.loads(request.body)
+        for entry in data:
+            serializer = PostRequestSerializer(data=entry)
```

Comment:
The serializer should be able to parse all the data, something like the following should do the trick.

You need to:

1. Parse all `request.data`
2. Tell the serializer to parse a list of items (`many=True`)
3. Raise an exception (which is automatically handled by DRF) if the data is wrong

```
serializer = PostSerializer(data=request.data, many=True)
serializer.is_valid(raise_exception=True)
```